modules/lib/src/main/resources/assets/js/v6/features/layout/AppShell/Sidebar.tsx

```diff
@@ -0,0 +1,242 @@
+import {Widget} from "@enonic/lib-admin-ui/content/Widget";
+import {Store} from "@enonic/lib-admin-ui/store/Store";
+import {LegacyElement} from "@enonic/lib-admin-ui/ui2/LegacyElement";
+import {CONFIG} from "@enonic/lib-admin-ui/util/Config";
+import {IconButton, Tooltip, Button, cn} from "@enonic/ui";
+import {CircleQuestionMark, Pen} from "lucide-react";
+import {AppContext} from "../../../../app/AppContext";
+import {ProjectIcon} from "../../../../app/ui2/list/ProjectIcon";
+import {ProjectContext} from "../../../../app/project/ProjectContext";
+
+type WidgetHandler = (widgetId: string) => void;
+
+type Props = {
+  name?: string;
+  widgets?: Widget[];
+  activeWidgetId?: string;
+  handler?: WidgetHandler;
```

Comment:
Too generic. Can't really understand what kind of handler it is? Keeping plain old `onClick` would be better